fineftp-server/src/ftp_session.cpp

```diff
@@ -485,6 +486,48 @@ namespace fineftp
     return;
   }
 
+  void FtpSession::handleFtpCommandSIZE(const std::string& param)
+  {
+    if (!logged_in_user_)
+    {
+      sendFtpMessage(FtpReplyCode::NOT_LOGGED_IN, "Not logged in");
+      return;
+    }
+    if (static_cast<int>(logged_in_user_->permissions_ & Permission::FileRead) == 0)
+    {
+      sendFtpMessage(FtpReplyCode::ACTION_NOT_TAKEN, "Permission denied");
+      return;
+    }
+
+    std::string local_path = toLocalPath(param);
+
+    std::ios::openmode open_mode =
+       std::ios::ate | (data_type_binary_ ? (std::ios::in | std::ios::binary) : (std::ios::in));
+    std::ifstream file(local_path, open_mode);
```

Comment:
The local_path must convert to an UTF16 wstring on windows

Reply:
Thanks. I will look into that.